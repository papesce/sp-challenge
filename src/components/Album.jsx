import React, { Component } from 'react'
import { GridListTile, GridListTileBar } from 'material-ui/GridList';
import { withStyles } from 'material-ui/styles';
import IconButton from 'material-ui/IconButton';
import InfoIcon from 'material-ui-icons/Info';
import ZoomInIcon from 'material-ui-icons/ZoomIn';


const styles = theme => ({
    tile: {
      margin: theme.spacing.unit,
      overflow: 'hidden',
      height: 300
    },
    
    
  });

export class Album extends Component {
  render() {
     const {classes, album, handleClick} = this.props;
    return (
       <GridListTile className={classes.tile}
         key={album.id} >
            <img src={album.images[1].url} alt={album.name} 
            onClick={handleClick} /> 
            <GridListTileBar
              title={album.name}
              subtitle={<span>by: {album.artists[0].name}</span>}
              actionIcon={<div>
                <IconButton href={album.external_urls.spotify} >
                  <InfoIcon color="rgba(255, 255, 255, 0.54)"/>
                </IconButton>
                <IconButton onClick={() => handleClick(album)} >
                  <ZoomInIcon color="rgba(255, 255, 255, 0.54)"/>
                </IconButton>
                </div>
              }
            />
          </GridListTile>
    )
  }
}

export default withStyles(styles)(Album)