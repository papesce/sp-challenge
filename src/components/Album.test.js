import React from 'react'
import Album from './Album'
import { shallow } from "enzyme";
import renderer from 'react-test-renderer';

describe('Album tests', () => {
    let album;
    beforeAll(() => { 
        album = {id:"x", name:"albumname", images: [{}, {url: "text"}],
        artists: [{name: "name"}], external_urls : {spotify: "eurl"}};
    })
    it('should ', () => {  
        shallow(<Album album={album}/>);
    })
    it('renders correctly', () => {
        const tree = renderer.create(<Album album={album}/>)
            .toJSON();
         expect(tree).toMatchSnapshot();
    });
    
})
