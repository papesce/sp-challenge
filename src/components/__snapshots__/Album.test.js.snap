// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album tests renders correctly 1`] = `
<li
  className="MuiGridListTile-root-2 Album-tile-1"
>
  <div
    className="MuiGridListTile-tile-3"
  >
    <img
      alt="album_name"
      src="text"
    />
    <div
      className="MuiGridListTileBar-root-6 MuiGridListTileBar-rootBottom-7 MuiGridListTileBar-rootWithSubtitle-9"
    >
      <div
        className="MuiGridListTileBar-titleWrap-10 MuiGridListTileBar-titleWrapActionRight-12"
      >
        <div
          className="MuiGridListTileBar-title-13"
        >
          <a
            href="eurl"
            style={
              Object {
                "color": "white",
              }
            }
          >
            album_name
          </a>
        </div>
        <div
          className="MuiGridListTileBar-subtitle-14"
        >
          <span>
            by: 
            name
          </span>
        </div>
      </div>
      <div
        className=""
      >
        <div>
          <button
            className="MuiButtonBase-root-26 MuiIconButton-root-17"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label-23"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-28 MuiIconButton-icon-24"
                color="rgba(255, 255, 255, 0.54)"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-29"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</li>
`;
